csrc/kernel_cache.cpp

```diff
@@ -201,7 +222,55 @@ std::vector<at::Tensor> FusionExecutorCache::runFusionWithInputs(
 
   KernelArgumentHolder args = prepareInputs(perm_inputs);
 
-  auto kernel_runtime = getKernelRuntimeFor(args, forced_index_type);
+  // Concretizing the fusion using args
+  auto fusion = fusion_.get();
+  if (has_dynamic_reshape_) {
+    ExpressionEvaluator expr_eval;
+
+    // Bind input scalars and tensor metadata to symbolic scalars
+    // Here we bind only the inputs that are needed to concretize dynamic
+    // transforms.
+    TORCH_CHECK(
+        args.size() == fusion->inputs().size(),
+        "Received ",
+        args.size(),
+        " inputs but expected ",
+        fusion->inputs().size());
+    for (auto i : c10::irange(args.size())) {
+      auto inpi = fusion->inputs()[i];
+      if (inpi->isIntegralScalar() &&
+          concretization_inputs_.find(inpi) != concretization_inputs_.end()) {
```

Comment:
Can we just bind all arguments no matter it's required for concretization or not?